### PR TITLE
Fixes #820 updated Component.updated function to check for Symbols

### LINKED ISSUE #820: Symbol props don't cause a re-render when they change
When a props name is hidden with a (native) symbol it doesnt seem to be getting picked up in the Component.updated check. The values are being passed in by skate.props but don't seem to be checked.

Will raise a PR today or tomorrow

### PR DESCRIPTION
**Pull request by lukebatchelor**
I had to skip the precommit hook because it just wasnt letting me commit. 
All the linting passes but it still complains about extraneous packages (even after pruning...)

I figured CI would pick up on any *actual* issues

### CODE REVIEW

File: src/api/component.js [hunk 1]
```diff
@@ -115,19 +116,23 @@ Component.extend = function extend(definition = {}, Base = this) {
...
-  for (const name in prev) {
-    if (prev[name] !== elem[name]) {
-      return true;
+  let wasUpdated = false;
+  // use get all keys so that we check Symbols as well as regular props
+  // using a for loop so we can break early
+  const allKeys = getAllKeys(prev);
+  for (let i = 0; i < allKeys.length; i += 1) {
+    if (prev[allKeys[i]] !== elem[allKeys[i]]) {
+      wasUpdated = true;
```
Comment (zzarcon) [inline src/api/component.js]: Why not `return true` here as it was before?

Comment (lukebatchelor) [inline src/api/component.js]: 😵 
I had it in a forEach before and returning true was returning from the forEach function. Completely missed that. Haha

File: src/api/component.js [hunk 2]
```diff
@@ -115,19 +116,21 @@ Component.extend = function extend(definition = {}, Base = this) {
...
 Component.updated = function updated(elem, prev) {
   if (!prev) {
     return true;
   }
-
-  // eslint-disable-next-line no-restricted-syntax
-  for (const name in prev) {
-    if (prev[name] !== elem[name]) {
+  // use get all keys so that we check Symbols as well as regular props
+  // using a for loop so we can break early
```
Comment (treshugart) [inline src/api/component.js]: In some perf testing I've done, `for` is also faster in performance critical code than `Array` methods. Especially in older browsers.

File: src/api/component.js [hunk 3]
```diff
@@ -115,19 +116,21 @@ Component.extend = function extend(definition = {}, Base = this) {
...
+  const allKeys = getAllKeys(prev);
+  for (let i = 0; i < allKeys.length; i += 1) {
```
Comment (treshugart) [inline src/api/component.js]: Might also be good to save `.length` as a `const` (`allKeysLen`, or something).

File: src/api/component.js [hunk 4]
```diff
@@ -115,19 +116,21 @@ Component.extend = function extend(definition = {}, Base = this) {
 // Skate
 //
 // This is a default implementation that does strict equality copmarison on
-// prevoius props and next props. It synchronously renders on the first prop
+// previous props and next props. It synchronously renders on the first prop
```
Comment (bqx) [inline src/api/component.js]: 👌

Comment (lukebatchelor) [inline src/api/component.js]: Boyscout 😎

**Comment by lukebatchelor** [comment]
LGTM

**Comment by jpnelson** [comment]
LGTM

**Comment by treshugart** [comment]
lgtm

**Pull request merged**